theia-python-docker/Dockerfile

```diff
@@ -1,22 +1,43 @@
 ARG NODE_VERSION=10
-FROM node:${NODE_VERSION}-stretch
+FROM node:${NODE_VERSION}
 
+# Install Python 3 from source
 RUN apt-get update \
-    && apt-get install -y python python-dev python-pip \
-    && apt-get clean && rm -rf /var/cache/apt/* && rm -rf /var/lib/apt/lists/* && rm -rf /tmp/*
+    && apt-get upgrade -y \
```

Comment:
Wow - I feel myself growing old as this builds and tests :)  Maybe there's at least a switch to skip the tests?

Reply:
> Wow - I feel myself growing old as this builds and tests :) Maybe there's at least a switch to skip the tests?

I saw the following comment from [stackoverflow](https://stackoverflow.com/questions/44708262/make-install-from-source-python-without-running-tests). I think there is a tradeoff between **build time** (skipping tests) and **better performance** in executing Python code.

At the moment I'm trying without the `--enable-optimizations` flag and it's building much more quickly.

Reply:
Maybe we can add a new build arg to control the behavior?

Reply:
Looks the speed improvement to be expected when using the `--enable-optimizations` build switch is [between 1 and 20%](https://bugs.python.org/issue24915) depending on the scenario. This is of course nice - if we were publishing a generic python it might be worth it. But for this image here I'll take the shorter built-time tradeoff.